src/components/favoritesButton.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+
+/**
+ * Improvements:
+ *  - animated heart icon for the favorites button (see: https://codepen.io/mapk/pen/ZOQqaQ)
+ *  - better background color
+ *  - include a toast (maybe "Added to Favorites") once clicked
+ *  - disable button as soon as it's clicked instead of having to refresh the page
+ *  - favorites button on the main nav bar should automatically requery for the
+ *    local storage data instead of having to do a manual refresh
+ *  - user can unfavorite immediately after they favorite. or like, any unfavoriting capability on this page
+ *  - relative positioning of button
+ */
+const FavoritesButton = (props) => {
+    let favoriteSpacesJson = JSON.parse(localStorage.getItem('favoriteSpaces'));
+
+    console.log(typeof favoriteSpacesJson.data.map(space => space.title).includes(props.space.title))
```

Comment:
the idea of this line of code here: `favoriteSpacesJson.data.map(space => space.title).includes(props.space.title)` can be very useful for checking for duplicates. Maybe you can try to use this to check if a space exists before adding it to recently viewed or favorites